### PR TITLE
Deprecate pullPolicy in favor of imagePullPolicy

### PR DESCRIPTION
**Pull request by bentsherman**
The `pollPolicy` pod option is an unnecessary shortcut for `imagePullPolicy` and should be removed in order to tighten up the nextflow / k8s interface. This PR simply deprecates `pullPolicy` so that users are warned that it may be removed in a future release.

### CODE REVIEW

File: modules/nextflow/src/test/groovy/nextflow/k8s/model/PodOptionsTest.groovy [hunk 1]
```diff
@@ -37,20 +37,20 @@ class PodOptionsTest extends Specification {
...
         when:
-        options = new PodOptions([ [pullPolicy:'Always'] ])
+        options = new PodOptions([ [imagePullPolicy: 'Always'] ])
         then:
         options.getImagePullPolicy() == 'Always'
 
         when:
-        options = new PodOptions([ [imagePullPolicy:'latest'] ])
+        options = new PodOptions([ [imagePullPolicy: 'latest'] ])
         then:
```
Comment (pditommaso) [inline modules/nextflow/src/test/groovy/nextflow/k8s/model/PodOptionsTest.groovy:54]: It should be kept a test also for `pullPolicy` to be sure it's still working

File: modules/nextflow/src/main/groovy/nextflow/k8s/K8sConfig.groovy [hunk 2]
```diff
@@ -55,11 +55,14 @@ class K8sConfig implements Map<String,Object> {
             this.podOptions.volumeClaims.add(new PodVolumeClaim(name, mount, subPath))
         }
 
-        // -- shortcut to pod image pull-policy
-        if( target.pullPolicy )
+        // -- shortcut to pod image pull policy
+        if( target.pullPolicy ) {
+            log.warn "The `pullPolicy` option is deprecated -- Use `imagePullPolicy` instead."
```
Comment (pditommaso) [inline modules/nextflow/src/main/groovy/nextflow/k8s/K8sConfig.groovy:60]: In principle, only the one in `PodOption` should be enough. It does not?

Comment (bentsherman) [reply modules/nextflow/src/main/groovy/nextflow/k8s/K8sConfig.groovy:60]: Apparently it is both a pod option and a `k8s` config option, so I deprecated both.

File: modules/nextflow/src/test/groovy/nextflow/k8s/model/PodOptionsTest.groovy (hunk 1 above)
Comment (bentsherman) [reply modules/nextflow/src/test/groovy/nextflow/k8s/model/PodOptionsTest.groovy:54]: Done

**Pull request closed without merging**